src/VectorBundle.jl

```diff
@@ -0,0 +1,461 @@
+
+Distance between vectors `X` and `Y` from the vector space at point `p`
+from the manifold `B.manifold`, that is the base manifold of `M`.
+"""
+distance(B::VectorBundleFibers, p, X, Y) = norm(B, p, X - Y)
+
+function get_basis(M::TangentBundleFibers, p, B::AbstractBasis{<:Any,TangentSpaceType})
+    return get_basis(M.manifold, p, B)
+end
+
+function get_coordinates(M::TangentBundleFibers, p, X, B::AbstractBasis)
+    return get_coordinates(M.manifold, p, X, B)
+end
+
+function get_coordinates!(M::TangentBundleFibers, Y, p, X, B::AbstractBasis)
+    return get_coordinates!(M.manifold, Y, p, X, B)
+end
+
+function get_vector(M::TangentBundleFibers, p, X, B::AbstractBasis)
+    return get_vector(M.manifold, p, X, B)
+end
+
+function get_vector!(M::TangentBundleFibers, Y, p, X, B::AbstractBasis)
+    return get_vector!(M.manifold, Y, p, X, B)
+end
+
+"""
+    injectivity_radius(M::TangentBundle)
+
+Injectivity radius of [`TangentBundle`](@ref) manifold is infinite if the base manifold
+is flat and 0 otherwise.
+See [https://mathoverflow.net/questions/94322/injectivity-radius-of-the-sasaki-metric](https://mathoverflow.net/questions/94322/injectivity-radius-of-the-sasaki-metric).
+"""
+function injectivity_radius(M::TangentBundle)
+    if is_flat(M.manifold)
+        return Inf
+    else
+        return 0.0
+    end
+end
+
+"""
+    inner(B::BundleFibers, p, X, Y)
+
+Inner product of vectors `X` and `Y` from the vector space of type `B.fiber`
+at point `p` from manifold `B.manifold`.
+"""
+inner(B::BundleFibers, p, X, Y)
+
+inner(B::TangentBundleFibers, p, X, Y) = inner(B.manifold, p, X, Y)
+@doc raw"""
+    inner(B::VectorBundle, p, X, Y)
+
+Inner product of tangent vectors `X` and `Y` at point `p` from the
+vector bundle `B` over manifold `B.fiber` (denoted $\mathcal M$).
+
+Notation:
+  * The point $p = (x_p, V_p)$ where $x_p ∈ \mathcal M$ and $V_p$ belongs to the
+    fiber $F=π^{-1}(\{x_p\})$ of the vector bundle $B$ where $π$ is the
+    canonical projection of that vector bundle $B$.
+  * The tangent vector $v = (V_{X,M}, V_{X,F}) ∈ T_{x}B$ where
+    $V_{X,M}$ is a tangent vector from the tangent space $T_{x_p}\mathcal M$ and
+    $V_{X,F}$ is a tangent vector from the tangent space $T_{V_p}F$ (isomorphic to $F$).
+    Similarly for the other tangent vector $w = (V_{Y,M}, V_{Y,F}) ∈ T_{x}B$.
+
+The inner product is calculated as
+
+$⟨X, Y⟩_p = ⟨V_{X,M}, V_{Y,M}⟩_{x_p} + ⟨V_{X,F}, V_{Y,F}⟩_{V_p}.$
+"""
+function inner(B::FiberBundle, p, X, Y)
+    px, Vx = submanifold_components(B.manifold, p)
+    VXM, VXF = submanifold_components(B.manifold, X)
+    VYM, VYF = submanifold_components(B.manifold, Y)
+    # for tangent bundle Vx is discarded by the method of inner for TangentSpaceAtPoint
+    # and px is actually used as the base point
+    return inner(B.manifold, px, VXM, VYM) + inner(FiberAtPoint(B.fiber, px), Vx, VXF, VYF)
+end
+
+function _inverse_retract(M::FiberBundle, p, q, ::FiberBundleInverseProductRetraction)
```

Comment:
here, the corresponding `inverse_retract` should be documented.

Reply:
This was removed from this PR.